Get some basic tests running

cottsay opened the pull request.

cottsay committed:
Add some basic testing

Signed-off-by: Scott K Logan <logans@cottsay.net>

clalancette reviewed (changes requested):
There's a few things I don't understand in this PR:

1.  It does a bunch of work to change `test_joy_twist.py`, but that test never actually gets run by the CMakeLists.txt.  Should we make it run in there?
1.  In my local testing, flake8 isn't happy with `test_joy_twist.py`; it is mostly complaining about import statements being in the wrong order and the use of double-quotes.  I think we'll want to fix those before merging.
1.  There are a bunch of changes to the copyright statement at the top of `test_joy_twist.py`, but then the copyright check is essentially disabled in CMakeLists.txt.  Either we should make the code changes to make the copyright check pass (and re-enable the test), or we should just disable that test (and thus we don't need the changes to the copyright).  I'd prefer the former if we can make it happen, but the latter is acceptable as well.

cottsay commented:
Thanks for taking a look, Chris.
1. It isn't being used. I ported that script to ROS 2 on a functional level, but I'm not sure how to integrate it. Guidance on the path forward here would be appreciated.
2. Sorry about that - you probably have a different flake8 than I do. I sorted the imports and switched to single quotes. (It did pass on my machine and on the buildfarm)
3. I changed the copyright because the lines were too long and it was upsetting flake8. The rest of the files are the reason I didn't enable that test. Adding copyrights seems like something we should do in ROS 1 as well.

clalancette commented:
>     1. It isn't being used. I ported that script to ROS 2 on a functional level, but I'm not sure how to integrate it. Guidance on the path forward here would be appreciated.

Ah, I see.  The major thing wrong with that test as it stands is that it does not use the launch_testing infrastructure.  In order for this test to work, it would need to bring that stuff in and then launch the unittest alongside an invocation of teleop_twist_joy.  One other thing to keep in mind if you go looking at this is that the "axes" field of the message *must* have the dead-man button enabled (button 5 by default), otherwise nothing will come out of teleop_twist_joy.  Once you have those things in place, I think this test would actually work.

> 
>     2. Sorry about that - you probably have a different flake8 than I do. I sorted the imports and switched to single quotes. (It did pass on my machine and on the buildfarm)

This is mostly fixed, though my flake8 was still complaining about the order of includes; basically "time" and "unittest" need to be before everything else.

> 
>     3. I changed the copyright because the lines were too long and it was upsetting flake8. The rest of the files are the reason I didn't enable that test. Adding copyrights seems like something we should do in ROS 1 as well.

All right, makes sense.  I'm fine with that change to make flake happy.  I agree about adding the copyrights to ROS 1, but it honestly isn't that urgent.

cottsay committed:
Fix one last flake8 violation

Signed-off-by: Scott K Logan <logans@cottsay.net>

cottsay committed:
Call cppcheck explicitly to specify LANGUAGE

cottsay committed:
Make all of the tests work with ROS 2

Two tests are disabled while we decided how to handle the difference in
behavior.

Signed-off-by: Scott K Logan <logans@cottsay.net>

cottsay committed:
Merge branch 'dashing' into tests

cottsay committed:
Add a missing test dependency

cottsay committed:
Update tests to reflect new ROS 2 turbo behavior

cottsay committed:
Add another test dependency

cottsay committed:
Stop running the default cppcheck

cottsay committed:
Make the test class names unique

cottsay commented on CMakeLists.txt:
```diff
@@ -46,28 +46,39 @@ install(DIRECTORY launch config
-  # add_rostest(test/only_turbo_joy.test)
-  # add_rostest(test/turbo_angular_enable_joy.test)
-  # add_rostest(test/turbo_angular_enable_joy_with_rosparam_map.test)
+  # Need to specify LANGUAGE because hpp extension isn't used
```
This is worth calling out - if the header used the `.hpp` extension instead of `.h`, we wouldn't need to call `ament_cppcheck` explicitly.

cottsay replied:
Nevermind, this trick doesn't work well enough for me to suppress the implicit invocation but still call cppcheck explicitly. Still, cppcheck could be run if the header were renamed.
I'm going to call it out-of-scope for this PR though.

clalancette replied:
Yeah, that's fine.  I'm going to to do a follow-up PR that does a bit more cleanup and enables composition.

cottsay committed:
Try re-arranging the ament_cppcheck call

cottsay committed:
Give up on getting cppcheck running with special args

clalancette committed:
Remove unnecessary shebang.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Change test_joy_twist.py to not have executable bits.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette reviewed (approved):
I made one small change, in that I removed the shebang from test_joy_twist.py; it's always imported and used, never run directly.  Other than that, it looks good to me so I'll merge this.  Thanks for iterating.

clalancette committed:
More test fixes.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
